kong/core/router.lua

```diff
@@ -493,6 +493,9 @@ function _M.new(apis)
 
 
     local host = headers["Host"] or headers["host"]
+    if host then
+      host = host:match("^([^:]+)") -- strip port number if given
```

Comment:
Better using `match(host, pattern)` in such a hot code path. Even better is to use ngx_pcre, to allow this bit to be JIT-compiled.